scipy/sparse/csgraph/_flow.pyx

```diff
@@ -0,0 +1,398 @@
+    graph constructed above contains this matrix as a block. Here's an example:
+
+    >>> graph = csr_matrix([[0, 1, 0, 1], [1, 0, 1, 0], [0, 1, 1, 0]])
+    >>> print(graph.toarray())
+    [[0 1 0 1]
+     [1 0 1 0]
+     [0 1 1 0]]
+    >>> i, j = graph.shape
+    >>> n = graph.nnz
+    >>> indptr = np.concatenate([[0],
+    ...                          graph.indptr + i,
+    ...                          np.arange(n + i + 1, n + i + j + 1),
+    ...                          [n + i + j]])
+    >>> indices = np.concatenate([np.arange(1, i + 1),
+    ...                           graph.indices + i + 1,
+    ...                           np.repeat(i + j + 1, j)])
+    >>> data = np.ones(n + i + j, dtype=np.int)
+    >>>
+    >>> graph_flow = csr_matrix((data, indices, indptr))
+    >>> print(graph_flow.toarray())
+    [[0 1 1 1 0 0 0 0 0]
+     [0 0 0 0 0 1 0 1 0]
+     [0 0 0 0 1 0 1 0 0]
+     [0 0 0 0 0 1 1 0 0]
+     [0 0 0 0 0 0 0 0 1]
+     [0 0 0 0 0 0 0 0 1]
+     [0 0 0 0 0 0 0 0 1]
+     [0 0 0 0 0 0 0 0 1]
+     [0 0 0 0 0 0 0 0 0]]
+
+    At this point, we can find the maximum flow between the added sink and the
+    added source and the desired matching can be obtained by restricting the
+    residual graph to the block corresponding to the original graph:
+
+    >>> flow = maximum_flow(graph_flow, 0, i+j+1)
+    >>> matching = flow.residual[1:i+1, i+1:i+j+1]
+    >>> print(matching.toarray())
+    [[0 1 0 0]
+     [1 0 0 0]
+     [0 0 1 0]]
+
+    This tells us that the first, second, and third vertex in :math:`U` are
+    matched with the second, first, and third vertex in :math:`V` respectively.
+
+    While this solves the maximum bipartite matching problem in general, note
+    that algorithms specialized to that problem will perform better. In
+    particular, :func:`maximum_bipartite_matching` will be faster when its
+    preconditions are met.
+
+    This approach can also be used to solve various common generalizations of
+    the maximum bipartite matching problem. If, for instance, some vertices can
+    be matched with more than one other vertex, this may be handled by
+    modifying the capacities of the new graph appropriately.
+
+    """
+    if not isspmatrix_csr(csgraph):
+        raise TypeError("graph must be in CSR format")
+    if not issubclass(csgraph.dtype.type, np.integer):
+        raise ValueError("graph capacities must be integers")
+    elif csgraph.dtype != ITYPE:
+        csgraph = csgraph.astype(ITYPE)
+    if source == sink:
+        raise ValueError("source and sink vertices must differ")
+    if source < 0 or source >= csgraph.shape[0]:
+        raise ValueError(f'source value ({source}) must be between 0 ' +
+                         f'and {csgraph.shape[0]-1}')
+    if sink < 0 or sink >= csgraph.shape[0]:
+        raise ValueError(f'sink value ({sink}) must be between 0 ' +
+                         f'and {csgraph.shape[0]-1}')
```

Comment:
No idea why tests on Python 3.5 are not failing, but this is invalid syntax. SciPy 1.4.0 will be the last release to support py35, so unfortunately this will need changing to `.format()` style strings

Reply:
Hm, yes, this should be covered by `test_raises_when_source_is_out_of_bounds` and `test_raises_when_sink_is_out_of_bounds`, and at least the [Python 3.5 output](https://travis-ci.org/scipy/scipy/jobs/609932536#L17296) on Travis states that it's 

> adding 'scipy/sparse/csgraph/tests/test_flow.py'

so unless I've missed something about test discovery, it's not that the tests are skipped altogether.

Reply:
I think I'm too lazy to try and reproduce that. I'm quite sure this can raise `SyntaxError` on py35, so let's just change it

Reply:
Yep, changed. We shouldn't dig into the green tests?